src/class-wc-gateway-btcpay.php

```diff
@@ -767,11 +827,24 @@ public function process_payment($order_id)
             }
 
             $order_total = $order->calculate_totals();
-            if (true === isset($order_total) && false === empty($order_total)) {
+            if (!empty($order_total)) {
                 $order_total = (float)$order_total;
```

Comment:
I kept the logic from before, not sure why it checks and forces a float value. When looking into the client library this gets even more strange, you can see if we pass an integer it will force to float internally anyway:
https://github.com/btcpayserver/btcpayserver-php-client/blob/master/src/BTCPayServer/Item.php#L110

Also strange if you pass a string it does `checkPriceFormat()` and throws an exception if it can't be converted to float:
https://github.com/btcpayserver/btcpayserver-php-client/blob/master/src/BTCPayServer/Item.php#L162

So no matter what we do here it will always end up as float or we need to change the client library but not sure if it is worth it (assuming it worked in the past)?

Reply:
ok thats fine then